lib/kubernetes-deploy/ejson_secret_provisioner.rb

```diff
@@ -0,0 +1,153 @@
+# frozen_string_literal: true
+require 'json'
+require 'base64'
+require 'open3'
+require 'kubernetes-deploy/kubeclient_builder'
+require 'kubernetes-deploy/logger'
+
+module KubernetesDeploy
+  class EjsonSecretError < FatalDeploymentError
+    def initialize(msg)
+      super("Creation of Kubernetes secrets from ejson failed: #{msg}")
+    end
+  end
+
+  class EjsonSecretProvisioner
+    include KubernetesDeploy::KubeclientBuilder
+
+    MANAGEMENT_ANNOTATION = "kubernetes-deploy.shopify.io/ejson-secret".freeze
+    MANAGED_SECRET_EJSON_KEY = "kubernetes_secrets".freeze
+    EJSON_SECRETS_FILE = "secrets.kubernetes-deploy.ejson".freeze
+    EJSON_KEYS_SECRET = "ejson-keys".freeze
+
+    def initialize(namespace:, context:, template_dir:, client: nil)
+      @namespace = namespace
+      @context = context
+      @ejson_file = "#{template_dir}/#{EJSON_SECRETS_FILE}"
+      @kubeclient = client || build_v1_kubeclient(@context)
+    end
+
+    def secrets_requested?
+      File.exist?(@ejson_file)
+    end
+
+    def create_secrets
+      raise EjsonSecretError, "#{EJSON_SECRETS_FILE} not found" unless secrets_requested?
+
+      with_decrypted_ejson do |decrypted|
+        secrets = decrypted[MANAGED_SECRET_EJSON_KEY]
+        unless secrets.present?
+          KubernetesDeploy.logger.warn("#{EJSON_SECRETS_FILE} does not have key #{MANAGED_SECRET_EJSON_KEY}. No secrets will be created.")
```

Comment:
It shouldn't hard fail because there could still be secrets that need to be pruned. It's mostly there in case they're deploying secrets for the first time and typo'd the key name, in which case hopefully they'd be on the lookout for logs about why it isn't working?